### PR TITLE
fix(alerting): Add support for client.insecure in email alerting provider

### LINKED ISSUE #581: Not able to send alert on outlook
### Describe the bug

I am trying to send alert on outlook but it is failing with this error: Failed to send an alert for endpoint=Liveness check: tls: failed to verify certificate: x509: certificate is valid for gwm.shashank.com, not skmailrelay.shashank.net
this is my configuration:
```
alerting:
  email:
    from: "skmailrelay.shashank.net"
    password: "abc"
    host: "skmailrelay.shashank.net"
    port: 25
    to: "abc@gmail.com"

endpoints:
  - name: Liveness check
    group: Gitlab
    url: "https://git.sk.abc.example.net/-/liveness"
    interval: 1m
    conditions:
      - "[STATUS] == 200"
      - "[BODY].status == ok" 
    alerts:
      - type: email
        send-on-resolved: true
        description: "health check failed"
```
Then I tried this:

```
alerting:
  email:
    from: "gwm.shashank.com"
    password: "abc"
    host: "gwm.shashank.com"
    port: 25
    to: "abc@gmail.com"

endpoints:
  - name: Liveness check
    group: Gitlab
    url: "https://git.sk.abc.example.net/-/liveness"
    interval: 1m
    conditions:
      - "[STATUS] == 200"
      - "[BODY].status == ok" 
    alerts:
      - type: email
        send-on-resolved: true
        description: "health check failed"
```
**Then the error is: Failed to send an alert for endpoint=Liveness check: dial tcp 209.141.168.113:25: i/o timeout**
I also tried this:
```
alerting:
  email:
    from: "skmailrelay.shashank.net"
    password: "abc"
    host: "skmailrelay.shashank.net"
    port: 25
    to: "abc@gmail.com"
    client:
      insecure: false

endpoints:
  - name: Liveness check
    group: Gitlab
    url: "https://git.sk.abc.example.net/-/liveness"
    interval: 1m
    conditions:
      - "[STATUS] == 200"
      - "[BODY].status == ok" 
    alerts:
      - type: email
        send-on-resolved: true
        description: "health check failed"
```

**but it seems this client:insecure functionality is not applicable to email alert type.**
so I tried this :

```
alerting:
  custom:
    from: "skmailrelay.shashank.net"
    password: "abc"
    host: "skmailrelay.shashank.net"
    port: 25
    to: "abc@gmail.com"    
    client:
      insecure: false
    body: |
      {
        "text": "[ALERT_TRIGGERED_OR_RESOLVED]: [ENDPOINT_GROUP] - [ENDPOINT_NAME] - [ALERT_DESCRIPTION]"
      }
endpoints:
  - name: Liveness check
    group: Gitlab
    url: "https://git.sk.abc.example.net/-/liveness"
    interval: 1m
    conditions:
      - "[STATUS] == 200"
      - "[BODY].status == ok" 
    alerts:
      - type: custom
        send-on-resolved: true
        description: "health check failed"
```
        
**then error is : invalid configuration**

### What do you see?

I have explained all the configurations and the errors in description.
**Last two configurations are violations I know, but I need help on the first two.** 
How and where it is checking ssl/tls, I also went through the codebase but wasn't able to find it.
Pls let me know if I am missing something.

### What do you expect to see?

Alert on outlook

### List the steps that must be taken to reproduce this issue

I have mentioned in description in detail

### Version

latest

### Additional information

pls look into this, i need to get it done ASAP

### PR DESCRIPTION
**Pull request by mxcd**
## Summary

This PR adds the possibility to configure the email client as `insecure` so that the TLS certificates of the mail server will not be checked.  
Default is `insecure: false`

Closes #581 

## Checklist

- [x] Tested and/or added tests to validate that the changes work as intended, if applicable.
- [x] Updated documentation in `README.md`, if applicable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -479,6 +479,7 @@ endpoints:
 | `alerting.email.overrides`         | List of overrides that may be prioritized over the default configuration                   | `[]`          |
 | `alerting.email.overrides[].group` | Endpoint group for which the configuration will be overridden by this configuration        | `""`          |
 | `alerting.email.overrides[].to`    | Email(s) to send the alerts to                                                             | `""`          |
+| `alerting.email.client.insecure`   | The email client does not verify the TLS certificate of the mail server                    | `false`       |
```
Comment (TwiN) [inline README.md:482]:
```suggestion
| `alerting.email.client.insecure`   | The email client does not verify the TLS certificate of the mail server                    | `false`       |
| `alerting.email.overrides`         | List of overrides that may be prioritized over the default configuration                   | `[]`          |
| `alerting.email.overrides[].group` | Endpoint group for which the configuration will be overridden by this configuration        | `""`          |
| `alerting.email.overrides[].to`    | Email(s) to send the alerts to                                                             | `""`          |
```

Comment (TwiN) [reply README.md:482]: Just moving `alerting.email.client.insecure` above the overrides

Comment (TwiN) [reply README.md:482]: also could you update the description of `alerting.email.client.insecure` to `Whether to skip TLS verification`?

File: README.md [hunk 2]
```diff
@@ -494,6 +495,10 @@ alerting:
     overrides:
       - group: "core"
         to: "recipient3@example.com,recipient4@example.com"
+    # You can configure the email client to skip TLS verification
+    # defaults to false
+    client:
+      insecure: false
```
Comment (TwiN) [inline README.md:501]:
```suggestion
    client:
      insecure: false
    overrides:
      - group: "core"
        to: "recipient3@example.com,recipient4@example.com"
```

**Review by TwiN — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mxcd) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (mxcd) [reply README.md]: done

**Review by TwiN — APPROVED** [review]

**Pull request merged**